pulser/waveforms.py

```diff
@@ -235,36 +240,36 @@ class CustomWaveform(Waveform):
             (in rad/µs). The number of samples dictates the duration, in ns.
     """
 
-    def __init__(self, samples):
+    def __init__(self, samples: np.ndarray):
```

Comment:
From looking at this, I think that sometimes it's a bit too restrictive to ask for a numpy array specifically. I think it's a good place to test using `ArrayLike` from `numpy.testing` (see [here](https://numpy.org/devdocs/reference/typing.html#numpy.typing.ArrayLike)). If it works, we can start doing that throughout the code.

Reply:
Just a reminder to look into this

Reply:
Also, `samples` can be `Parametrized` too.